### PR TITLE
DOCS File migration changes for 4.4.0

### PR DESCRIPTION
**Pull request by chillu**
See https://github.com/silverstripe/silverstripe-versioned/issues/177
Merge alongside with https://github.com/silverstripe/silverstripe-assets/pull/223

### CODE REVIEW

File: docs/en/02_Developer_Guides/14_Files/03_File_Security.md [hunk 1]
```diff
@@ -186,7 +182,7 @@ The urls for these two files, however, do not reflect the physical structure dir
   up the content of the hidden file, conditional on a permission check.
 
 When the file `NewCompanyLogo.gif` is made public, the url will not change, but the file location
```
Comment (maxime-rainville) [inline docs/en/02_Developer_Guides/14_Files/03_File_Security.md]: The URL use to serve the file does change. When serving a protected file, you will get a hash path.

File: docs/en/02_Developer_Guides/14_Files/04_File_Storage.md [hunk 2]
```diff
@@ -53,32 +53,100 @@ storage backend to determine how variants are managed.
...
+It leaves out the `.protected/` folder, and leaves that to SilverStripe's integrated routing:
+`http://www.example.com/assets/my-protected-folder/b63923d8d4/my-protected-file.jpg`.
 
 For more information on how protected files are stored see the [file security](/developer_guides/files/file_security)
 section.
 
+## Versioned file paths
+
+Older versions of file contents are kept in the `.protected` folder,
+following the same rules as [protected file paths](#protected-file-paths).
```
Comment (maxime-rainville) [inline docs/en/02_Developer_Guides/14_Files/04_File_Storage.md]: That kind of make it sounds like this is the default behaviour, which it isn't. It's only going to do that `keep_archived_assets` is enabled.

Comment (chillu) [reply docs/en/02_Developer_Guides/14_Files/04_File_Storage.md]: Good catch, fixed

File: docs/en/02_Developer_Guides/14_Files/03_File_Security.md (hunk 1 above)
Comment (chillu) [reply docs/en/02_Developer_Guides/14_Files/03_File_Security.md]: Reworded a bit

File: docs/en/02_Developer_Guides/14_Files/03_File_Security.md [hunk 3]
```diff
@@ -9,7 +9,7 @@ in your system. As pages and dataobjects can be either versioned, or restricted
 members, it is necessary at times to apply similar logic to any files which are attached to these objects
 in the same way.
 
-Out of the box SilverStripe comes with an two asset stores, a public and a protected one.
+Out of the box, SilverStripe comes with an two asset stores: a public and a protected one.
```
Comment (michalkleiner) [inline docs/en/02_Developer_Guides/14_Files/03_File_Security.md]: Out of the box, SilverStripe comes with ~an~ two asset stores: a public and a protected one.

Comment (maxime-rainville) [reply docs/en/02_Developer_Guides/14_Files/03_File_Security.md]:
```suggestion
Out of the box, SilverStripe comes with two asset stores: a public and a protected one.
```

**Pull request merged**